src/dbnode/storage/bootstrap/bootstrapper/persist.go

```diff
@@ -94,6 +94,14 @@ func PersistBootstrapIndexSegment(
 	retentionOpts := ns.Options().RetentionOptions()
 	nowFn := resultOpts.ClockOptions().NowFn()
 	earliestRetentionTime := retention.FlushTimeStart(retentionOpts, nowFn())
+
+	// If bootstrapping is taking more time than our retention period, we might end up in a situation
+	// when earliestRetentionTime is larger than out block end time. This means that the blocks
+	// got outdated during bootstrap so we just skip building index segments for them.
+	if earliestRetentionTime.After(blockEnd) {
+		return result.IndexBlock{}, fs.ErrOutOfRetentionClaim
```

Comment:
This error (`ErrOutOfRetentionClaim`) is taken from `index_claims_manager.go`, its message is `out of retention index volume claim`. I'm not sure if it is appropriate to reuse it here as this code path is not related to index claims.

Reply:
The reason I decided to reuse this error is because the callers were already handling it in most of the places, introducing new error type would force them to implement additional checks. Also the newly introduced error would probably be very similar so it might get confusing. 
I would probably suggest to rename the existing error to something more general like `ErrOutOfRetention` so it could be used in different contexts more easily.